utils/model_utils.py

```
@@ -379,47 +382,63 @@
-                save_strategy="no",
+                eval_strategy=eval_strategy,
+                save_strategy=IntervalStrategy.NO,
                 output_dir=output_directory,
                 seed=training_seed,
                 data_seed=data_seed,
             )
             trainer = Trainer(
                 model_init=lambda: AutoModelForSequenceClassification.from_pretrained(
                     model_name,
-                    num_labels=len(dataset.unique("label")),
+                    num_labels=len(train_dataset.unique("label")),
```

Ruff (0.8.2)

Function definition does not bind loop variable `train_dataset`

(B023)